test/lib/ufe/testAttributeBlock.py

```diff
@@ -261,5 +261,91 @@ def testAttributeBlocking3dCommonApi(self):
         # authoring new transformation edit is allowed.
         self.assertTrue(mayaUsdUfe.isAttributeEditAllowed(rotateAttr))
 
+    def testAttributeBlocking3dMatrixOps(self):
+        '''
+        Verify authoring transformation attribute(s) in weaker layer(s) are not permitted if there exist opinion(s) 
+        in stronger layer(s) when using UsdTransform3dMatrixOp.
+        '''
+
+        # create new stage
+        cmds.file(new=True, force=True)
+
+        # Open usdCylinder.ma scene in testSamples
+        mayaUtils.openCylinderScene()
+
+        # get the stage
+        proxyShapes = cmds.ls(type="mayaUsdProxyShapeBase", long=True)
+        proxyShapePath = proxyShapes[0]
+        stage = mayaUsd.lib.GetPrim(proxyShapePath).GetStage()
+
+        # cylinder prim
+        cylinderPrim = stage.GetPrimAtPath('/pCylinder1')
+        self.assertIsNotNone(cylinderPrim)
+
+         # create 3 sub-layers ( LayerA, LayerB, LayerC ) and set the edit target to LayerB.
+        rootLayer = stage.GetRootLayer()
+        subLayerC = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerC")[0]
+        subLayerB = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerB")[0]
+        subLayerA = cmds.mayaUsdLayerEditor(rootLayer.identifier, edit=True, addAnonymous="SubLayerA")[0]
+
+        # check to see the sublayers added
+        addedLayers = [subLayerA, subLayerB, subLayerC]
+        self.assertEqual(rootLayer.subLayerPaths, addedLayers)
+
+        # set the edit target to LayerB
+        cmds.mayaUsdEditTarget(proxyShapePath, edit=True, editTarget=subLayerB)
+
+        # create a xformable and give it a matrix4d xformOp:transform
+        # This will match the UsdTransform3dMatrixOp, but not the Maya API.
+        cylinderXformable = UsdGeom.Xformable(cylinderPrim)
+        transformOp = cylinderXformable.AddTransformOp()
+        xform = Gf.Matrix4d(1.0).SetTranslate(Gf.Vec3d(10, 20, 30))
+        transformOp.Set(xform)
+        self.assertTrue(cylinderPrim.GetPrim().HasAttribute("xformOp:transform"))
```

Comment:
Line 298 to 304: I create a `matrix4d xformOp:transform` with the initial translate values set to `[10, 20, 30]` to match the 3dMatrixOp.